lib/chef/search/query.rb

```diff
@@ -88,8 +88,9 @@ def search(type, query='*:*', *args, &block)
 
         if block
           response["rows"].each { |row| block.call(row) if row }
-          unless (response["start"] + response["rows"].length) >= response["total"]
-            args_h[:start] = response["start"] + response["rows"].length
+          next_start = response["start"] + (args_h[:rows] || response["rows"].length)
+          unless next_start >= response["total"]
```

Comment:
good point, this is a good place to make a note that the results may not be the same size as the amount of rows we asked for.